OData/test/UnitTest/System.Web.OData.Test/OData/Builder/LinkGenerationHelpersTest.cs

```diff
@@ -88,6 +89,41 @@ public void GenerateNavigationLink_WorksToGenerateExpectedNavigationLink_ForSing
             Assert.Equal(expectedNavigationLink, uri.AbsoluteUri);
         }
 
+        [Theory]
+        [InlineData(false, "http://localhost/MyOrders(42)/OrderLines(21)/OrderLines")]
```

Comment:
For a model is not with Foos/Bars/Boos, but with Foos/Bars, will the code also generate Foos(key)/Bars(key)/Bars?